src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs

```diff
@@ -355,6 +355,22 @@ private void AssertConnectionState(SubscriptionConnectionServerMessage connectio
                     throw new SubscriptionDoesNotExistException(
                         $"Subscription With Id '{_options.SubscriptionName}' cannot be opened, because it does not exist. " + connectionStatus.Exception);
                 case SubscriptionConnectionServerMessage.ConnectionStatus.Redirect:
+                    if (_options.Strategy == SubscriptionOpeningStrategy.WaitForFree)
+                    {
+                        if (connectionStatus.Data != null && connectionStatus.Data.TryGetMember(
+                                nameof(SubscriptionConnectionServerMessage.SubscriptionRedirectData.StartRegisterConnection), out object startRegisterConnectionObject))
+                        {
+                            if (startRegisterConnectionObject is long startRegisterConnection)
+                            {
+                                if (TimeSpan.FromTicks(DateTime.UtcNow.Ticks - startRegisterConnection) >= _options.MaxErroneousPeriod)
```

Comment:
This assumes client and server are using the same clock, no?
What is the actual issue here that youa re trying to solve? That we timeout waiting for a connection?

Reply:
I use `DateTime.UtcNow` in both cases.

In situation there is a subscription with `WaitForFree` connection strategy and 2 workers, 
The actual issue is:
1. the 1st worker got `SubscriptionDoesNotBelongToNodeException` and marked `_lastConnectionFailure` = now
2. the 2nd worker got the connection
3. the 1st worker waits until the connection is free for a long time (longer than `MaxErroneousPeriod` of the worker)
4. the connection is free and the 1st worker starts to process the subscription and gets `SubscriptionDoesNotBelongToNodeException` again
5. the 1st worker tries to reconnect and try to `AssertLastConnectionFailure`, in `AssertLastConnectionFailure` we check if the  `_lastConnectionFailure` (which was set in (1)) now it will be more then `MaxErroneousPeriod`, but all that time was spent in waiting for free connection and wasn't actually failing.

Reply:
I believe that Oren was referring to a possible clock desynchronization between client and server machines, e.g. client has 1:00 PM UTC while on the server we have 1:05 UTC. Not that they have different time zone set.

Reply:
Yes, that is the issue

Reply:
I will return the timespan it took waiting until the connection is free then